dev/profile_startup.dart

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env dart
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:async';
+import 'dart:convert';
+
+const int ITERATIONS = 3;
+
+String runWithLoggingSync(List<String> cmd, {
+  bool checked: true,
+  String workingDirectory
+}) {
+  ProcessResult results =
+      Process.runSync(cmd[0], cmd.getRange(1, cmd.length).toList(), workingDirectory: workingDirectory);
+  if (results.exitCode != 0) {
+    String errorDescription = 'Error code ${results.exitCode} '
+        'returned when attempting to run command: ${cmd.join(' ')}';
+    print(errorDescription);
+    if (results.stderr.length > 0)
+      print('Errors logged: ${results.stderr.trim()}');
+    if (checked)
+      throw errorDescription;
+  }
+  if (results.stdout.trim().isNotEmpty)
+    print(results.stdout.trim());
+  return results.stdout;
+}
+
+double timeToFirstFrame(trace) {
+  // FIXME: Sort! Trace events are not guarenteed to be in timestamp order.
+  List events = trace['traceEvents'];
+  int firstTimeStamp = events[0]['ts'].toInt();
+  var firstSwap = events.firstWhere((e) => e['name'] == 'NativeViewGLSurfaceEGL:RealSwapBuffers');
+  int swapStart = firstSwap['ts'].toInt();
+  int swapEnd = swapStart + firstSwap['dur'].toInt();
+  return (swapEnd - firstTimeStamp) / 1000; // microseconds to milliseconds.
+}
+
+Future<double> test(String tracesDir, String projectPath, int runNumber) async {
+  // If we used package:path we could grab the basename of project_path
+  // and include that in the trace_name.
```

Comment:
if we get to that point we should just make this a feature of the `flutter` tool